components/emailoctopus/emailoctopus.app.mjs

```diff
@@ -1,11 +1,70 @@
+import { axios } from "@pipedream/platform";
+import constants from "./common/constants.mjs";
+
 export default {
-    type: "app",
-    app: "emailoctopus",
-    propDefinitions: {},
-    methods: {
-      // this.$auth contains connected account data
-      authKeys() {
-        console.log(Object.keys(this.$auth));
+  type: "app",
+  app: "emailoctopus",
+  propDefinitions: {
+    listId: {
+      type: "string",
+      label: "List",
+      description: "List where the source will be performed.",
+      async options({ page }) {
+        const { data } = await this.getLists({
+          page: page + 1,
+        });
+
+        return data.map((list) => ({
+          label: list.name,
+          value: list.id,
+        }));
       },
     },
-  };
\ No newline at end of file
+  },
+  methods: {
+    _getParams(params) {
+      const resObj = {
+        "api_key": `${this.$auth.api_key}`,
+      };
+      for (const key in params) {
+        resObj[key] = params[key];
+      }
+      return resObj;
+    },
+    async _makeRequest({
+      $, url, path, params, ...otherConfig
+    }) {
+      const baseUrl = `${constants.BASE_URL}${constants.VERSION_PATH}`;
+
+      const config = {
+        url: url || `${baseUrl}${path}`,
+        params: this._getParams(params),
+        ...otherConfig,
+      };
+      return axios($ || this, config);
+    },
+    async getLists({ page }) {
+      return this._makeRequest({
+        method: "GET",
+        path: "/lists",
+        params: {
+          page,
+        },
+      });
+    },
+    async getContacts({ listId }) {
+      return this._makeRequest({
+        method: "GET",
+        path: `/lists/${listId}/contacts`,
+      });
+    },
```

Comment:
According to [this doc](https://emailoctopus.com/api-documentation/lists/get-all-contacts), all results are not returned, where `page` defaults to `1` and `limit` defaults to `100`. Is it possible to have more than `100` contacts? If so can we just make sure that we are returning all contacts here? [This](https://github.com/PipedreamHQ/pipedream/blob/5a4392ae161a9f60ecda6cd501240b5af87f09a8/components/gorgias/gorgias.app.mjs#L127) can be a good example.